src/test/java/org/junit/tests/assertion/AssertionTest.java

```diff
@@ -83,11 +98,29 @@ public void arraysActualNullMessage() {
     }
 
     @Test
-    public void arraysDifferentLengthMessage() {
+    public void arraysDifferentLengthDifferingAtStartMessage() {
+        try {
+            assertArrayEquals("not equal", (new Object[]{true}), (new Object[]{false, true}));
+        } catch (AssertionError exception) {
+            assertEquals("not equal: array lengths differed, expected.length=1 actual.length=2; arrays first differed at element [0]; expected:<true> but was:<false>", exception.getMessage());
+        }
```

Comment:
Funny you should say that. I initially wrote the tests in the way you suggest, then noticed that none of the other tests in this class have that pattern. Rather, for every message test, there is a corresponding failure test that checks that the assertion really does fail. On the one hand, that does leave the message tests brittle, but on the other hand, it means each test only tests one thing - either the fact that the assertion fails, or the message it produces.

My tests follow the existing pattern. I'm happy to change them to break the pattern, but perhaps it would be better to refactor all of the tests in this class to work that way.